### DISCUSSION TITLE
filter items shown in form repeater?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#12896] [Help]
### Package

Form builder

### Package Version

v3.2.80

### How can we help you?

I have a form for let's say a house.  
All the data for that house is in a single JSON. So, no eloquent relationship.  
The form has a repeater for each of the rooms.

Is it possible to filter which rooms are shown in the repeater?  
E.g. only show the rooms of `type: bedroom`.

**Comment by wivaku** [comment]
I may have found a solution, but feedback is welcome.

```
Forms\Components\Repeater::make('rooms')
    ->afterStateHydrated(fn($component, $state) => $component->state(
        collect($state)
            ->where('type', 'bedroom')
            ->values()
            ->toArray()
        )
    )
    ->schema([...])
```

**Reply by KONEY** [reply]
Hello,
my feedback is that it does work but in a destructive way. For example, adding a second repeater with a different filter (es: type=suite) it will filter items of both the first and second repeater by type=suite, ending with the same values.
Hope its helpful to someone to save some time :)

**Comment by Chaaarles** [comment]
I had the same problem today, and found another, potentially better, solution.
You can use the `relationship` function of `Repeater` to modify the query fetching relationships.

Here is an example from my project where I need to filter based on an enum of types.
```PHP
Repeater::make('quiz_grades')
                    ->relationship('grades', fn (Builder $query) => $query->where('type', GradeType::Quiz))
```

In your case, I believe something like this should work:
```PHP
Repeater::make('bedrooms')
    ->relationship('rooms', fn (Builder $query) => $query->where('type', 'bedroom'))
```
